tasks/internal/process-file-updates-task/process-file-updates-task.yaml

```diff
@@ -290,50 +283,30 @@ spec:
         # Remove trailing newline
         openMRList=$(echo "$openMRList" | sed '/^$/d')
 
-        if [ -n "$openMRList" ]; then        
+        if [ -n "$openMRList" ]; then
             while IFS= read -r oneItem; do
                 mrNum=$(echo "$oneItem" | cut -f1 | tr -d '!')
-                foundMR=false
                 glab mr diff "${mrNum}" --repo "${UPSTREAM_REPO}" > "${TEMP}"/oneMR.diff 
-
-                if [[ ! -s "${TEMP}"/lineFile ]]; then
-                    # only seed exists and no replacements  
-                    grep '^[+][^+]' "${TEMP}"/oneMR.diff | sed -E 's/^[+]//' | tee "${TEMP}/mrFile"
-
-                    # compare if the contents and the updated file names are the same 
-                    if  diff -q "${TEMP}"/mrFile "${targetFile}" > /dev/null; then
-                        grep "^diff --git" "${TEMP}"/oneMR.diff | tee "${TEMP}"/tmpFile
-                        while read -r line ; do
-                            changedFileName=$(echo "$line" | awk '{sub(/^b\//, "", $NF); print $NF}')
-                            if [[ "${changedFileName}" != "${targetFile}" ]]; then
-                                continue
-                            fi
-                            foundMR=true
-                            break
-                        done < "${TEMP}"/tmpFile
-                    fi
-                else
-                    # replacements exist
-                    grep '^[-+@]' "${TEMP}"/oneMR.diff | sed -E 's/^[@]//; s/@@.*//' | tee "${TEMP}/mrFile"
-                    # remove "a/" in  "--- a/dir/file" and "b/" in "+++ b/dir/file" 
-                    grep '^[-+@]' "${TEMP}"/tempMRFile.diff | sed -E 's/^[@]//; s/@@.*//' | \
-                        sed -E 's/^(---|\+\+\+) [ab]\//\1 /' | tee "${TEMP}/tmpFile"
-                    if diff -q "${TEMP}"/mrFile "${TEMP}"/tmpFile > /dev/null; then
-                        foundMR=true
-                    fi
-                fi
-                # if all the lines are matched, the MR is the same one
-                # it should exit 0 if the same MR exists 
-                if [[ "$foundMR" == true ]]; then
-                    echo "there is an existing MR with the same updates in the repo"
+               
+                # remove lines like '--- /dev/null'
+                grep '^[-+@]' "${TEMP}"/oneMR.diff | sed -E 's/^[@]//; s/@@.*//' | \
+                    grep -v '\-\-\-' | tee "${TEMP}/mrFile"
+
+                # remove "a/" in  "--- a/dir/file" and "b/" in "+++ b/dir/file"
+                grep '^[-+@]' "${TEMP}"/tempMRFile-cached.diff | sed -E 's/^[@]//; s/@@.*//' | \
+                    sed -E 's/^(---|\+\+\+) [ab]\//\1 /'| grep -v '\-\-\-' | \
+                        tee "${TEMP}/tmpFile"
```

Comment:
Why is this needed? I just tried it myself and the output of both `glab mr diff` and `git diff --cached` was exactly the same. Both showed exactly the same metadata and when I did `diff` on those two outputs, there was no difference.

Reply:
It's different. The output of `glab mr diff` is like 
```
--- file1
+++ file1
```
And the out put of  `git diff --cached` is like 
```
--- a\file1
+++ b\file1
```

Reply:
OK, sorry, I made a mistake in my test :) You're right. What can help with that is `git diff --cached --no-prefix`. I still get some differences, but perhaps at least the a/b handling can be removed?
```
$ diff <(git diff --cached --no-prefix) <(glab mr diff 11 --repo gitlab.cee.redhat.com/rhtap-release/advisories)
1,2d0
< diff --git data/advisories/assisted-installer-tenant/2025/2724/advisory.yaml data/advisories/assisted-installer-tenant/2025/2724/advisory.yaml
< index 68bfde2c3..f1b84eb52 100644
14,17c12
< diff --git schema/advisory-copy.json schema/advisory-copy.json
< new file mode 100644
< index 000000000..271ec6c25
< --- /dev/null
---
> --- schema/advisory-copy.json
```

Reply:
No matter. I have handled the differences in the script.

Reply:
Is that ok? or do you think it's better to use "--no-prefix" in the command?

Reply:
Actually, I think you had a `git fetch` of the mr branch in a previous version, right? Why not use that?
```
$ git fetch rhtap-release merge-requests/11/head
$ git diff --cached FETCH_HEAD
```
That displays zero changes for me right now. One could say that you pay extra for the fetch, but the `glab mr diff` command probably isn't much cheaper.

If that works, I think it would be the most elegant way - without having to manipulate the diff. WDYT?

Reply:
Sounds good. I will use `git fetch`. Thanks

Reply:
Updated!

Reply:
Looks good, thanks!